test/lib.api.v1.settings-controller.test.js

```diff
@@ -4,19 +4,78 @@ var request = require('supertest');
 var settingsController = require('../lib/api/v1/settings-controller');
 var botMaster = require('../lib/fetching/bot-master');
 var Source = require('../models/source');
+var config = require('../config/secrets');
+var _ = require('underscore');
 
 describe('Settings controller', function() {
   before(function(done) {
     botMaster.kill();
     botMaster.addListeners('source', Source.schema);
     botMaster.addListeners('fetching', settingsController);
-    
+
     Source.create({nickname: 'one', media: 'dummy', keywords: 'one'});
     Source.create({nickname: 'two', media: 'dummy', keywords: 'two'});
     Source.create({nickname: 'three', media: 'dummy', keywords: 'three'});
-    
-    var config = require('../config/secrets');
-    config.updateFetching(false, function(err){
+
+    twitterSettings =  _.clone(config.get().twitter);
+    testSettings = {
+      consumer_key: 'testing',
+      consumer_secret: 'test',
+      access_token: 'api',
+      access_token_secret: 'error',
+      configured: false,
+      on: false,
+    };
+
+    config.updateFetching(false, function(err) {
+      done();
+    });
+  });
+
+  describe('GET /api/v1/settings/twitter', function() {
+    it('should return the settings JSON of twitter', function(done) {
+      request(settingsController)
+        .get('/api/v1/settings/twitter')
+        .expect(200, {twitter: config.get({ reload: true }).twitter, setting: 'twitter' }, done);
+    });
+  });
+
+  describe('POST /api/v1/settings/media/twitter', function() {
```

Comment:
I wanted to make sure that in the long run, nobody changes the meaning of the POST method to actually post a new setting, rather than just testing the setting. Maybe better to make a GET request with an extra test toggle?